src/runtime/arch_tinygowasm.go

```diff
@@ -84,8 +84,23 @@ func libc_calloc(nmemb, size uintptr) unsafe.Pointer {
 
 //export realloc
 func libc_realloc(ptr unsafe.Pointer, size uintptr) unsafe.Pointer {
-	runtimePanic("unimplemented: realloc")
-	return nil
+	newMalloc := libc_malloc(size)
+	if newMalloc == nil || ptr == nil {
+		return newMalloc
+	}
```

Comment:
Please call `alloc` directly. It won't return a nil pointer.

```suggestion
	newMalloc := alloc(size)
```

Reply:
I saw that -gc=conservative never returns nil, but was not sure if it can be assumed for other gcs. 

Regarding ptr I was trying to stick to POSIX and the musl implementation:
> If ptr is a null pointer, realloc() shall be equivalent to malloc() for the specified size.
https://pubs.opengroup.org/onlinepubs/9699919799/